File: JAVA-CarRacingProject/ProiectCurse/src/main/java/repository/interfaces/IInscriereRepository.java
package repository.interfaces;


import domain.Inscriere;
import repository.ICrudRepository;

public interface IInscriereRepository extends ICrudRepository<Integer, Inscriere> {
    /**
     * @param idParticipant - idParticipant dupa care se filtreaza Inscrierile
     * @return toate Inscrierile filtrate
     */
    Iterable<Inscriere> findInscrieribyIdParticipant(Integer idParticipant);

    /**
     * @param idCursa - dupa care se filtreaza Inscrierile
     * @return toate Inscrierile filtrate
     */
    Iterable<Inscriere> findInscrieribyIdCursa(Integer idCursa);
}

File: C#-CarRacingProject/Repository/Interfaces/IInscriereRepository.cs
using ProiectCSharpVersion.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProiectCSharpVersion.Repository.Interfaces
{
    interface IInscriereRepository:ICrudRepository<int,Inscriere>
    {
        //filtrare dupa idParticipant
        IEnumerable<Inscriere> GetInscrieriByIdParticipant(int idParticipant);

        //filtrare dupa idCUrsa
        IEnumerable<Inscriere> GetInscrieriByIdCursa(int idCursa);
    }
}
